docs/src/main/md/index.md

```diff
@@ -0,0 +1,8780 @@
+    return jdbcTemplate.queryForList("SELECT * FROM user;");
+}
+```
+
+You can rely on [Spring Boot data source
+auto-configuration](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html#boot-features-connect-to-production-database)
+to configure a `DataSource` bean. In other words, properties like the
+SQL username, `spring.datasource.username`, and password,
+`spring.datasource.password` can be used. There is also some
+configuration specific to Google Cloud SQL (see "Cloud SQL Configuration
+Properties" section below).
+
+|                                       |                     |          |                                                                        |
+| ------------------------------------- | ------------------- | -------- | ---------------------------------------------------------------------- |
+| Property name                         | Description         | Required | Default value                                                          |
+| `spring.datasource.username`          | Database username   | No       | MySQL: `root`; PostgreSQL: `postgres`                                  |
+| `spring.datasource.password`          | Database password   | No       | `null`                                                                 |
+| `spring.datasource.driver-class-name` | JDBC driver to use. | No       | MySQL: `com.mysql.cj.jdbc.Driver`; PostgreSQL: `org.postgresql.Driver` |
+
+<div class="note">
+
+If you provide your own `spring.datasource.url`, it will be ignored,
+unless you disable Cloud SQL auto configuration with
+`spring.cloud.gcp.sql.enabled=false` or
+`spring.cloud.gcp.sql.jdbc.enabled=false`.
+
+</div>
+
+#### `DataSource` creation flow
+
+Spring Boot starter for Google Cloud SQL registers a
+`CloudSqlEnvironmentPostProcessor` that provides a correctly formatted
+`spring.datasource.url` property to the environment based on the
+properties mentioned above. It also provides defaults for
+`spring.datasource.username` and `spring.datasource.driver-class-name`,
+which can be overridden. The starter also configures credentials for the
+JDBC connection based on the properties below.
+
+The user properties and the properties provided by the
+`CloudSqlEnvironmentPostProcessor` are then used by [Spring
+Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html)
+to create the `DataSource`. You can select the type of connection pool
+(e.g., Tomcat, HikariCP, etc.) by [adding their dependency to the
+classpath](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html#boot-features-connect-to-production-database).
+
+Using the created `DataSource` in conjunction with Spring JDBC provides
+you with a fully configured and operational `JdbcTemplate` object that
+you can use to interact with your SQL database. You can connect to your
+database with as little as a database and instance names.
+
+## R2DBC Support
```

Comment:
This needs to be indented one more level, so it's under Cloud SQL (this was also the problem in adoc; I just noticed it here).